### PR TITLE
Created test setup environment and fixed test(s)

### PR DESCRIPTION
**Pull request by at88mph**
I added a Gradle build as well.

### CODE REVIEW

**Comment by heyitsanthony** [comment]
Won't AGPL conflict with jetcd's [apache license](https://github.com/coreos/jetcd/blob/master/LICENSE)?

**Comment by at88mph** [comment]
I think all of the dependencies use the Apache License.  What uses AGPL?

File: src/test/java/com/coreos/jetcd/DockerCommandRunner.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
```
Comment (heyitsanthony) [inline src/test/java/com/coreos/jetcd/DockerCommandRunner.java]: AGPL

Comment (at88mph) [inline src/test/java/com/coreos/jetcd/DockerCommandRunner.java]: I see now.  That was a mistake as my IDE has a template header for new files.  I will remove those headers and fix Travis.

File: src/main/java/com/coreos/jetcd/EtcdClusterImpl.java [hunk 2]
```diff
@@ -8,11 +8,13 @@
 /**
  * Implementation of cluster client
  */
-public class EtcdClusterImpl implements EtcdCluster {
+public class EtcdClusterImpl implements EtcdCluster
+{
```
Comment (AdoHe) [inline src/main/java/com/coreos/jetcd/EtcdClusterImpl.java]: why update this? problem with code formatter plugin?

File: src/test/java/com/coreos/jetcd/AbstractEtcDClusterTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import com.coreos.jetcd.integration.EtcdInstance;
+import com.coreos.jetcd.integration.ExternalInstance;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+
+abstract class AbstractEtcDClusterTest extends AbstractTest
```
Comment (AdoHe) [inline src/test/java/com/coreos/jetcd/AbstractEtcDClusterTest.java]: `AbstractEtcdClusterTest`

File: src/test/java/com/coreos/jetcd/AbstractEtcDInstanceTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.util.logging.Logger;
+
+
+/**
+ * Test to provide running instances for tests, if necessary.
+ *
+ * Set the ENDPOINTS System property if you have a running instance already.
+ */
+abstract class AbstractEtcDInstanceTest extends AbstractTest
```
Comment (AdoHe) [inline src/test/java/com/coreos/jetcd/AbstractEtcDInstanceTest.java]: the same.

File: src/test/java/com/coreos/jetcd/DockerCommandRunner.java [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+
+/**
+ * Simple command runner for Docker.
+ */
+public class DockerCommandRunner
+{
```
Comment (AdoHe) [inline src/test/java/com/coreos/jetcd/DockerCommandRunner.java]: I would suggest put this in test package.

**Comment by AdoHe** [comment]
@at88mph thanks very much, I will do a deep review this weekend.

**Comment by lburgazzoli** [comment]
@at88mph what is the purpose of gradle in addition to maven here? Anything not possible with maven? 

I think we should use only one build system.

**Comment by at88mph** [comment]
I agree.  I was trying to get a handle on why the tests seemed to run in parallel, and I was testing to see if Maven was the issue since they failed every three or four runs.

I will remove the Gradle files.



Dustin N. Jenkins 
Tel/Tél: 250.217.4970 | facsimile/télécopieur: 250.363.0045 | Dustin.Jenkins@nrc-cnrc.gc.ca <mailto:Dustin.Jenkins@nrc-cnrc.gc.ca> 
National Research Council Canada | 5071 West Saanich Rd, Victoria BC. V9E 2E7 
Conseil national de recherches Canada | 5071, ch. West Saanich, Victoria (C.-B) V9E 2E7 
Government of Canada | Gouvernement du Canada

**Comment by fanminshi** [comment]
closing the testing framework is added.

**Pull request closed without merging**